src/CharityVault.sol

```diff
@@ -143,96 +185,166 @@ contract CharityVault is ERC20, Auth {
 
         // TODO: this needs to be updated to calculate how much use should get
         // Transfer underlying tokens to the sender.
-        underlying.safeTransfer(msg.sender, underlyingAmount);
+        UNDERLYING.safeTransfer(msg.sender, underlyingAmount);
 
         emit CharityWithdraw(msg.sender, underlyingAmount);
     }
 
     // TODO: Charity Withdraw function
     // TODO: this function should only be callable by the charity
 
-    /// @notice Burns fcvTokens and sends underlying tokens to the charity.
-    /// @param amount The amount of fcvTokens to redeem for underlying tokens.
+    /// @notice Burns rcvTokens and sends underlying tokens to the charity.
+    /// @param amount The amount of rcvTokens to redeem for underlying tokens.
     function charityWithdraw(uint256 amount) external {
         // Query the vault's exchange rate.
         uint256 exchangeRate = exchangeRateCurrent();
 
         // TODO: we have to somehow keep track of how much is owed to the charity vs the user
-        // Convert the amount of fcvTokens to underlying tokens.
-        // This can be done by multiplying the fcvTokens by the exchange rate.
-        uint256 underlyingAmount = ((exchangeRate * amount) / 10**decimals) * (feePercent / 100.0);
+        // Convert the amount of rcvTokens to underlying tokens.
+        // This can be done by multiplying the rcvTokens by the exchange rate.
+        uint256 underlyingAmount = ((exchangeRate * amount) / 10**decimals) * (BASE_FEE / 100.0);
 
-        // Burn inputed fcvTokens.
-        _burn(charity, amount);
+        // Burn inputed rcvTokens.
+        _burn(CHARITY, amount);
 
-        // If the withdrawal amount is greater than the float, pull tokens from Fuse.
-        // if (underlyingAmount > getFloat()) vault.pullIntoFloat(underlyingAmount);
+        // Get the user's underlying balance
+        uint256 user_balance = VAULT.balanceOfUnderlying(msg.sender);
```

Comment:
idt u want the user's balance here the vault is custodying vault shares on behalf of the them